### PR TITLE
SMS login ☎️

### PR DESCRIPTION
**Pull request by cjdenio**
This ✨ magic PR ✨ adds Twilio-powered SMS login!

the details:

**model changes:**

`User` has a fancy new (optional) `phone` field that must be an E.164-formatted number (e.g. +16031234567)

**some environment variables:**
```
TWILIO_ACCOUNT_SID=<>
TWILIO_AUTH_TOKEN=<>
TWILIO_NUMBER=<E.164-formatted number registered to your Twilio account>
```

**shiny new endpoint:**

`/v1/users/phone_auth` works the same as `/v1/users/auth`, except it takes a `phone` parameter rather than `email` (and doesn't automatically create a user if none exists)

**low-effort screenshot:**

(I hardly poured my life's work into this, so this can be changed/closed/merged with zero impact on my feelings)

### CODE REVIEW

**Comment by sampoder** [comment]
Caleb this is so cool

**Comment by zachlatta** [comment]
omg omg omg omg

--
Zach Latta
https://zachlatta.com <http://zachlatta.com>
@zachlatta <https://twitter.com/zachlatta> / fb
<https://www.facebook.com/crynix> / github <https://github.com/zachlatta>

**Comment by zachlatta** [comment]
Can someone take point on this? I'm not going to be able to review and
merge this with everything on my plate right now. I am 100% in support of
getting this merged and adding phone auth to all auth systems, especially
Bank

--
Zach Latta
https://zachlatta.com <http://zachlatta.com>
@zachlatta <https://twitter.com/zachlatta> / fb
<https://www.facebook.com/crynix> / github <https://github.com/zachlatta>

File: app/controllers/v1/users_controller.rb [hunk 1]
```diff
@@ -38,6 +38,34 @@ def auth
       end
     end
 
+    def phone_auth
+      user = User.find_by(phone: params[:phone])
```
Comment (motdotla) [inline app/controllers/v1/users_controller.rb:42]: @cjdenio if params[:phone] is nil here is this going to find a User in the db where the `phone` field is null?

Comment (cjdenio) [reply app/controllers/v1/users_controller.rb]: Great catch, should be fixed now

**Comment by cjdenio** [comment]
Hmm, I guess [this PR](https://github.com/hackclub/api/pull/454) is replacing this one?

**Comment by zachlatta** [comment]
Yeah, that’s the plan. Maybe they can pull some of your code? We’re
currently working on SMS auth and adding support for multiple active
sessions at once

--
Zach Latta
https://zachlatta.com <http://zachlatta.com>
@zachlatta <https://twitter.com/zachlatta> / fb
<https://www.facebook.com/crynix> / github <https://github.com/zachlatta>

**Pull request closed without merging**